compiler/src/diagnostics/lenses.re

```diff
@@ -0,0 +1,77 @@
+open Grain_typed;
+
+[@deriving yojson]
+type lens_t = {
+  line: int,
+  signature: string,
+};
+
+[@deriving yojson]
+type lens_list_t = {lenses: list(lens_t)};
+
+let lenses_to_json_string = (lenses: list(lens_t)): string => {
+  let lensList: lens_list_t = {lenses: lenses};
+  let json_list = lens_list_t_to_yojson(lensList);
+  Yojson.Basic.pretty_to_string(Yojson.Safe.to_basic(json_list));
+};
+
+let output_lenses = (program: Typedtree.typed_program): list(lens_t) => {
+  let lenses =
+    List.map(
+      (statement: Typedtree.toplevel_stmt) => {
+        let d = statement.ttop_desc;
+        switch (d) {
+        | TTopExpr(expr) =>
+          let buf = Buffer.create(64);
+          let ppf = Format.formatter_of_buffer(buf);
+          let t: Types.type_expr = expr.exp_type;
+          Printtyp.type_expr(ppf, t);
+          Format.pp_print_flush(ppf, ());
+          let signt = Buffer.contents(buf);
```

Comment:
I _believe_ this can be cleaned up with some of the utilities in `Format`, but I don't have it on-hand. Let's land it with this because it is working and we can always do a cleanup pass when I find those utils.